### PR TITLE
Example for pubsub authenticated push

### PR DESCRIPTION
**Pull request by ajaaym**
Add example for pubsub authenticated push

### CODE REVIEW

**Comment by ajaaym** [comment]
@anguillanneuf Can you please review and help with Java 8 CI error log?

File: appengine-java8/pubsub/README.md [hunk 1]
```diff
@@ -53,6 +53,21 @@ gcloud beta pubsub subscriptions create <your-subscription-name> \
   --ack-deadline 30
 ```
 
+- Create a subscription for authenticated pushes to send messages to a Google Cloud Project URL such as https://<your-project-id>.appspot.com/authenticated-push.
+
+The push auth service account must have Service Account Token Creator Role assigned, which can be done in the Cloud Console [IAM & admin](https://console.cloud.google.com/iam-admin/iam) UI.
+`--push-auth-token-audience` is optional. If set, remember to modify the audience field check in `PubSubAuthenticatedPush.java` (line 36).
```
Comment (lesv) [inline appengine-java8/pubsub/README.md]: You can provide a link to the file if you wish.
`[PubSubAuthenticatedPush.java](..../PubSubAuthenticatedPush.java#L36)`

File: appengine-java8/pubsub/src/main/java/com/example/appengine/pubsub/PubSubAuthenticatedPush.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet(value = "/pubsub/authenticated-push")
+public class PubSubAuthenticatedPush extends PubSubPush {
+  private final String pubsubVerificationToken = System.getenv("PUBSUB_VERIFICATION_TOKEN");
+  private final MessageRepository messageRepository;
+  private final GoogleIdTokenVerifier verifier =
+      new GoogleIdTokenVerifier.Builder(new NetHttpTransport(), new JacksonFactory())
+          .setAudience(Collections.singletonList("example.com"))
```
Comment (lesv) [inline appengine-java8/pubsub/src/main/java/com/example/appengine/pubsub/PubSubAuthenticatedPush.java]: You may wish to comment here why it should be changed and refer back to the README

File: appengine-java8/pubsub/src/main/java/com/example/appengine/pubsub/PubSubHome.java [hunk 3]
```diff
@@ -45,5 +65,15 @@ private static String convertToHtmlTable(List<Message> messages) {
     return sb.toString();
   }
 
-  private PubSubHome() { }
+  private static String convertStringsToHtmlTable(List<String> strings) {
```
Comment (lesv) [inline appengine-java8/pubsub/src/main/java/com/example/appengine/pubsub/PubSubHome.java]: I'm ok w/ this, but it really should be template based or anti-template based. https://github.com/agentgt/jatl

File: appengine-java8/pubsub/src/main/java/com/example/appengine/pubsub/PubSubAuthenticatedPush.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdTokenVerifier;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import java.io.IOException;
+import java.util.Collections;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
```
Comment (anguillanneuf) [inline appengine-java8/pubsub/src/main/java/com/example/appengine/pubsub/PubSubAuthenticatedPush.java]: We'll need to add region tags (for [docs](https://cloud.google.com/pubsub/docs/push#authentication_and_authorization_by_the_push_endpoint)) and comments for this section. The region tags should be `[START gae_standard_pubsub_auth_push]` and `[END gae_standard_pubsub_auth_push]`. The comments are the same as in the Node.js/Python sample. Thanks!

Comment (anguillanneuf) [reply appengine-java8/pubsub/src/main/java/com/example/appengine/pubsub/PubSubAuthenticatedPush.java]: I didn't select the right region for my comment. Should be line 30 - 74.

File: appengine-java8/pubsub/src/main/java/com/example/appengine/pubsub/PubSubAuthenticatedPush.java (hunk 2 above)
Comment (ajaaym) [reply appengine-java8/pubsub/src/main/java/com/example/appengine/pubsub/PubSubAuthenticatedPush.java]: done

File: appengine-java8/pubsub/src/main/java/com/example/appengine/pubsub/PubSubHome.java (hunk 3 above)
Comment (ajaaym) [reply appengine-java8/pubsub/src/main/java/com/example/appengine/pubsub/PubSubHome.java]: done

File: appengine-java8/pubsub/src/main/java/com/example/appengine/pubsub/PubSubAuthenticatedPush.java (hunk 4 above)
Comment (ajaaym) [reply appengine-java8/pubsub/src/main/java/com/example/appengine/pubsub/PubSubAuthenticatedPush.java]: done

File: appengine-java8/pubsub/src/main/java/com/example/appengine/pubsub/PubSubAuthenticatedPush.java [hunk 5]
```diff
@@ -20,30 +20,46 @@
...
+import java.util.Base64;
 import java.util.Collections;
+import java.util.stream.Collectors;
 import javax.servlet.ServletException;
 import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+// [START gae_flex_pubsub_auth_push]
```
Comment (anguillanneuf) [inline appengine-java8/pubsub/src/main/java/com/example/appengine/pubsub/PubSubAuthenticatedPush.java]: This is for [GAE Standard](https://github.com/GoogleCloudPlatform/java-docs-samples/tree/1c394688a61815ed14d74b23ce1f49ae77be3517/appengine-java8/pubsub) actually. Please use `gae_standard_pubsub_auth_push`.

Comment (ajaaym) [reply appengine-java8/pubsub/src/main/java/com/example/appengine/pubsub/PubSubAuthenticatedPush.java]: done

**Pull request merged**